### PR TITLE
Fix parsing certain postfix expressions with top level collection literals

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Partial fix for https://github.com/dotnet/roslyn/issues/68766.  More changes incoming in future PRs.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@333fred @RikkiGibson can you take a look since Chuck is on vacation?

File: src/Compilers/CSharp/Test/Syntax/Parsing/StatementAttributeParsingTests.cs [hunk 1]
```diff
@@ -5512,14 +5512,18 @@ void Goo(int i)
...
+                {
+                    void Goo(int i)
+                    {
+                        [A]++i;
+                    }
+                }
+                """,
+                // (5,14): error CS1002: ; expected
+                //         [A]++i;
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "i").WithLocation(5, 14));
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Test/Syntax/Parsing/StatementAttributeParsingTests.cs:5526]: this was an existing error recovery case that i added a few years back (recovering when attributes were put in incorrect places).  It's still an error after this pr, just one that views the above as an invalid operation on a collection-expr, not an invalid attribute on an expression statement.  Trying to detect this case and keep the old parsing isn't worth it IMO.

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 2]
```diff
@@ -7440,8 +7440,15 @@ private SyntaxList<AttributeListSyntax> ParseStatementAttributeDeclarations()
             while (this.CurrentToken.Kind == SyntaxKind.OpenBracketToken)
                 ParseBracketedArgumentList();
 
-            // Check the next token to see if it indicates the `[...]` sequence we have is a term or not. 
-            var isCollectionExpression = this.CurrentToken.Kind is SyntaxKind.DotToken or SyntaxKind.QuestionToken or SyntaxKind.ExclamationToken;
+            // Check the next token to see if it indicates the `[...]` sequence we have is a term or not. This is the
+            // same set of tokens that ParsePostFixExpression looks for.
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs:7444]: are we going to remember to make a corresponding change here if another postfix operator is ever added to the language? :)

File: src/Compilers/CSharp/Test/Syntax/Parsing/CollectionLiteralParsingTests.cs [hunk 3]
```diff
@@ -10609,6 +10609,289 @@ public void MemberAccess22A()
         EOF();
     }
 
+    [Fact]
+    public void MemberAccess23()
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Syntax/Parsing/CollectionLiteralParsingTests.cs:10613]: consider naming these tests something like PostFixOperator

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/Parser/LanguageParser.cs:7444]: I'll add comment

File: src/Compilers/CSharp/Test/Syntax/Parsing/CollectionLiteralParsingTests.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Syntax/Parsing/CollectionLiteralParsingTests.cs:10613]: Will fix in next pr

**Pull request merged**